packages/components/input/src/use-input.ts

```diff
@@ -145,9 +145,13 @@ export function useInput<T extends HTMLInputElement | HTMLTextAreaElement = HTML
   const isFilledByDefault = ["date", "time", "month", "week", "range"].includes(type!);
   const isFilled = !isEmpty(inputValue) || isFilledByDefault;
   const isFilledWithin = isFilled || isFocusWithin;
-  const baseStyles = clsx(classNames?.base, className, isFilled ? "is-filled" : "");
+  const isHiddenType = type === "hidden";
   const isMultiline = originalProps.isMultiline;
 
+  const baseStyles = clsx(classNames?.base, className, isFilled ? "is-filled" : "", {
+    hidden: isHiddenType,
```

Comment:
Move this to the input styles in the `theme` package, use the `data-hidden` attr

Reply:
The reason is because tailwindcss only compiles the styles inside the `theme` package not inside the `react` or `input` one